base/binaryplatforms.jl

```diff
@@ -0,0 +1,999 @@
+incompatibilities yet, bless its heart).  In reality, this actually checks for symbols
+within LLVM, but that is close enough for our purposes, as you can't mix configurations
+between Julia and LLVM; they must match.
+"""
+function detect_cxxstring_abi()
+    # First, if we're not linked against libstdc++, then early-exit because this doesn't matter.
+    libstdcxx_paths = filter(x -> occursin("libstdc++", x), Libdl.dllist())
+    if isempty(libstdcxx_paths)
+        # We were probably built by `clang`; we don't link against `libstdc++`` at all.
+        return nothing
+    end
+
+    function open_libllvm(f::Function)
+        for lib_name in ("libLLVM", "LLVM", "libLLVMSupport")
+            hdl = Libdl.dlopen_e(lib_name)
+            if hdl != C_NULL
+                try
+                    return f(hdl)
+                finally
+                    Libdl.dlclose(hdl)
+                end
+            end
+        end
+        error("Unable to open libLLVM!")
+    end
+
+    return open_libllvm() do hdl
+        # Check for llvm::sys::getProcessTriple(), first without cxx11 tag:
+        if Libdl.dlsym_e(hdl, "_ZN4llvm3sys16getProcessTripleEv") != C_NULL
+            return "cxx03"
+        elseif Libdl.dlsym_e(hdl, "_ZN4llvm3sys16getProcessTripleB5cxx11Ev") != C_NULL
+            return "cxx11"
+        else
+            @warn("Unable to find llvm::sys::getProcessTriple() in libLLVM!")
+            return nothing
+        end
+    end
+end
+
+"""
+    host_triplet()
+
+Build host triplet out of `Sys.MACHINE` and various introspective utilities that
+detect compiler ABI values such as `libgfortran_version`, `libstdcxx_version` and
+`cxxstring_abi`.  We do this without using any `Platform` tech as it must run before
+we have much of that built.
+"""
+function host_triplet()
+    str = Sys.MACHINE
+    libgfortran_version = detect_libgfortran_version()
+    if libgfortran_version !== nothing
+        str = string(str, "-libgfortran", libgfortran_version.major)
+    end
+
+    libstdcxx_version = detect_libstdcxx_version()
+    if libstdcxx_version !== nothing
+        str = string(str, "-libstdcxx", libstdcxx_version.patch)
+    end
+
+    cxxstring_abi = detect_cxxstring_abi()
+    if cxxstring_abi !== nothing
+        str = string(str, "-", cxxstring_abi)
+    end
+
+    # Add on julia_version extended tag
+    str = string(str, "-julia_version+", VersionNumber(VERSION.major, VERSION.minor, VERSION.patch))
+
+    return str
+end
+
+# Cache the host platform value, and return it if someone asks for just `Platform()`.
+default_host_platform = HostPlatform(parse(Platform, host_triplet()))
+"""
+    Platform()
+
+Return the `Platform` object that corresponds to the current host system, with all
+relevant comparison strategies set to host platform mode.  This is equivalent to:
+
+    HostPlatform(parse(Platform, Base.BinaryPlatforms.host_triplet()))
+"""
+function Platform()
+    global default_host_platform
```

Comment:
```suggestion
```

I think?

It seems a bit odd for me that this is called `Platform` but it isn't a constructor.

Reply:
Yes, I struggled with this.  I think we should perhaps instead call this `HostPlatform()`.